### PR TITLE
Create clockwise_spiral.cpp

### LINKED ISSUE #26: How to print clockwise spiral traversal of a binary tree
**MicroByte Title:** BinaryTree_LevelOrderTraversal_Types

**Improvement Proposal:** 
- Add solution to clockwise spiral traversal of binary tree as mentioned in Further Exploration Question 3
- Implement it in any language and add briefly documented code

**Benefits of proposed improvements:** 

- The learner will be able to find a solution to the Further Exploration section too

### PR DESCRIPTION
**Pull request by verma-pulkit**
## Closes #26 

- [ ] Added/updated activities for learners
- [ ] Didn't give the solutions to the added/updated activities straight away
- [ ] Followed a LearnByDoing approach
- [x] Clean and well documented code (if source code files are involved)
- [x] Added images for better understanding if deemed necessary

### CODE REVIEW

**Comment by Ak-Shaw** [comment]
@verma-pulkit Kindly follow the contribution guidelines. If you are updating a MicroByte by any chance, you need to follow [this](https://github.com/Crio-Bytes/DSA/blob/main/.github/PULL_REQUEST_TEMPLATE/updating-microbyte.md) template for it. Kindly edit your PR description for the same.

**Review by Ak-Shaw — CHANGES_REQUESTED** [review]
@verma-pulkit It seems that you can improve your work by refactoring your code in the main method and segregate it into helper methods. That way you won't have a hefty main method, you'll have a more easily manageable one. 

Also I'll suggest you add image(s) for the code in work, just for reference. Take a look in the MicroByte and you'll have a clear idea for that.

File: BinaryTree_LevelOrderTraversal_Types/src/clockwise_spiral.cpp [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                trav.push(currnode->right);
+        }
+        col = -1;
+    }
+
+    // for (int i=0; i<height; i++){
+    //     cout<<endl;
+    //     for (int j=0; j<width; j++)
+    //         cout<<elements[i][j]<<"  ";
+    // }
```
Comment (amoghrajesh) [inline BinaryTree_LevelOrderTraversal_Types/src/clockwise_spiral.cpp:130]: Please remove this if it is not needed

**Review by amoghrajesh — CHANGES_REQUESTED** [review]
Hey @verma-pulkit this is some good code you have written.
I have a few things where you can improve ->
1.  Let your main file just have the `creation of a tree` and calling a function to do the task ie) `print clockwise spiral`
2. Get rid of the commented code which is not used (This is an important practice which you will appreciate in a long run)
3. Your final output shows as ->`10 , 24 , 23 , 22 , 21 , 12 , 13 , 15 , 14 ,` . Get rid of the additional comma in the end 
4. Also as @Ak-Shaw mentioned, follow the template while submitting the PR

**Review by Ak-Shaw — APPROVED** [review]
@verma-pulkit Good work. LGTM! :rocket:

**Review by amoghrajesh — APPROVED** [review]
Hey, @verma-pulkit really good work, man. 
I am happy you took all the feedback and applied it well.
I hope you enjoyed doing it too 🥇

**Comment by amoghrajesh** [comment]
@all-contributors kindly add @verma-pulkit for:

- [x] code
- [x] content

**Comment by verma-pulkit** [comment]
Hey..thanks for the appreciation. Really happy working with you.

**Pull request merged**